Jetcaster/app/src/main/java/com/example/jetcaster/ui/MainActivity.kt

```diff
@@ -30,10 +40,30 @@ class MainActivity : ComponentActivity() {
         // This app draws behind the system bars, so we want to handle fitting system windows
         WindowCompat.setDecorFitsSystemWindows(window, false)
 
+        /**
+         * Flow of [FoldableInfo] that emits every time there's a change in the windowLayoutInfo
+         */
+        val foldableInfo = windowInfoRepository().windowLayoutInfo
+            .flowWithLifecycle(this.lifecycle)
+            .map { layoutInfo ->
+                val foldingFeature: FoldingFeature? =
+                    layoutInfo.displayFeatures.find { it is FoldingFeature } as? FoldingFeature
+                FoldableInfo(
```

Comment:
When the user runs this app on a device with no folding feature, there will still be a non-null `FoldableInfo` instance. Have you considered creating a sealed class `FoldableInfo` for this, with two subclasses `NoInfo` and `Info(...}`?

Reply:
For now, I think it's fine. But as a consumer of the flow I'd still be confused that I'm actually getting a FoldableInfo instance, even though my emulator isn't foldable at all 🤷